guides/operator/orca-redis-to-sql.md

```diff
@@ -0,0 +1,195 @@
+
+GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, REFERENCES, INDEX, ALTER, LOCK TABLES, EXECUTE, SHOW VIEW ON `orca`.* TO 'orca_migrate'@'%'
+```
+
+When Orca starts up, it will perform database migrations to ensure its running the correct schema.
+
+### Amazon Aurora
+
+At Netflix, we use Amazon Aurora MySQL and have configured it with the following settings, missing values are defaults.
+Your mileage may vary.
+
+#### Aurora Parameter Groups
+
+- *binlog_cache_size*: `32768`
+- *default_tmp_storage_engine*: `InnoDB`
+- *general_log*: `0`
+- *innodb_adaptive_hash_index*: `0`
+- *innodb_buffer_pool_size*: `{DBInstanceClassMemory*3/4}`
+- *key_buffer_size*: `16777216`
+- *log_queries_not_using_indexes*: `0`
+- *log_throttle_queries_not_using_indexes*: `60`
+- *long_query_time*: `0.5`
+- *max_allowed_packet*: `25165824`
+- *max_binlog_size*: `134217728`
+- *query_cache_size*: `{DBInstanceClassMemory/24}`
+- *query_cache_type*: `1`
+- *read_buffer_size*: `262144`
+- *slow_query_log*: `1`
+- *sync_binlog*: `1`
+- *tx_isolation*: `READ-COMMITTED`
+
+#### Aurora DB Parameters Group
+
+- *binlog_checksum*: `NONE`
+- *binlog_error_action*: `IGNORE_ERROR`
+- *binlog_format*: `MIXED`
+- *character_set_client*: `utf8mb4`
+- *character_set_connection*: `utf8mb4`
+- *character_set_database*: `utf8mb4`
+- *character_set_filesystem*: `utf8mb4`
+- *character_set_results*: `utf8mb4`
+- *character_set_server*: `utf8mb4`
+- *collation_connection*: `utf8mb4_unicode_ci`
+- *collation_server*: `utf8mb4_unicode_ci`
+- *innodb_checksums*: `0`
+
+#### MariaDB
+
+We use the MariaDB JDBC driver for connecting to Aurora MySQL. The MariaDB driver is Aurora cluster aware, which takes care of automatic master failover operations. Due to licensing issues, we can't ship Orca with the MariaDB driver.
+
+An example of wiring up MariaDB into Orca can be found here: [robzienert/orca-mariadb-extension](https://github.com/robzienert/orca-mariadb-extension).
+
+## Configuring Orca for SQL
+
+This configuration is your baseline for Orca to talk to SQL, in orca.yml:
+
+```yaml
+sql:
+  enabled: true
+  connectionPool:
+    jdbcUrl: jdbc:mysql://localhost:3306/orca
+    user: orca_service
+    password: hunter2
+    connectionTimeout: 5000
+    maxLifetime: 30000
+    # MariaDB-specific:
+    maxPoolSize: 50
+  migration:
+    jdbcUrl: jdbc:mysql://localhost:3306/orca
+    user: orca_migrate
+    password: hunter2
+
+# Ensure we're only using SQL for accessing execution state
+executionRepository:
+  sql:
+    enabled: true
+  redis:
+    enabled: false
+
+# Reporting on active execution metrics will be handled by SQL
+monitor:
+  activeExecutions:
+    redis: false
+```
+
+## Migrating from SQL to Redis
+
+Migrating without downtime from Redis to SQL is a three step process:
+
+1. Deploy Orca with the `DualExecutionRepository` writing to both Redis and SQL.
+2. Deploy a new Orca cluster with migrators enabled and queue processing disabled.
```

Comment:
Is it possible to deploy only one orca that has both migrators enabled as well as queue processing to simplify the migration behavior?

Reply:
This does work but has not been extensively tested. The migrators can be memory demanding and some users would potentially need to run with a more generous heap to successfully do all the things without impact.